sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/PreviousHospitalizationSteps.java

```diff
@@ -57,6 +63,67 @@ public PreviousHospitalizationSteps(
           ComparisonHelper.compareEqualEntities(collectedHospitalization, previousHospitalization);
         });
+
+    And(
+        "^I check if Previous hospitalization Popup contains additional fields$",
+        () -> {
+          webDriverHelpers.isElementVisibleWithTimeout(ADMITTED_AS_INPATIENT, 1);
+          webDriverHelpers.isElementVisibleWithTimeout(DATE_OF_ISOLATION, 1);
+          webDriverHelpers.clickOnWebElementBySelector(DISCARD_BUTTON);
+        });
+
+    And(
+        "I set Isolation as {string}",
+        (String isolationOption) ->
+            webDriverHelpers.clickWebElementByText(ISOLATION_OPTIONS, isolationOption));
+
+    Then(
+        "^I check the edited and saved current hospitalization is correctly displayed in previous hospitalization window$",
+        () -> {
+          createdHospitalization = hospitalizationService.generateHospitalization();
+          webDriverHelpers.clickOnWebElementBySelector(FIRST_PREVIOUS_HOSPITALIZATION_ENTRY);
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(REGION_COMBOBOX);
+          PreviousHospitalization collectedHospitalization = collectPreviousHospitalizationData();
+          softly.assertEquals(
+              collectedHospitalization.getDateOfVisitOrAdmission(),
+              createdHospitalization.getDateOfVisitOrAdmission(),
+              "Date of visit or admission is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getDateOfDischargeOrTransfer(),
+              createdHospitalization.getDateOfDischargeOrTransfer(),
+              "Date of discharge or transfer is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getReasonForHospitalization(),
+              createdHospitalization.getReasonForHospitalization(),
+              "Reason for hospitalization is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getSpecifyReason(),
+              createdHospitalization.getSpecifyReason(),
+              "Specify reason is not correct.");
+
+          softly.assertEquals(
+              collectedHospitalization.getStayInTheIntensiveCareUnit(),
+              createdHospitalization.getStayInTheIntensiveCareUnit(),
+              "Stay in the intensive care unit is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getStartOfStayDate(),
+              createdHospitalization.getStartOfStayDate(),
+              "Start of stay date is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getEndOfStayDate(),
+              createdHospitalization.getEndOfStayDate(),
+              "End of stay date is not correct");
+
+          softly.assertEquals(
+              collectedHospitalization.getIsolation(),
+              createdHospitalization.getIsolation(),
+              "Isolation is not correct");
+        });
```

Comment:
`    softly.assertAll();` is needed after a set of assertions. Otherwise they won't be performed.